src/ui/SimpleFilter/SimpleFilter.ts

```diff
@@ -0,0 +1,259 @@
+  };
+
+  public checkboxes: { checkbox: Checkbox, value: string }[] = [];
+  public checkboxContainer: Dom;
+  private expanded = false;
+  private circleElement: Dom;
+
+  constructor(public element: HTMLElement, public options: ISimpleFilterOptions, public bindings?: IComponentBindings) {
+    super(element, SimpleFilter.ID, bindings);
+    this.options = ComponentOptions.initComponentOptions(element, SimpleFilter, options);
+
+    const select = $$('span', { className: 'coveo-simplefilter-select' });
+    const selectText = $$('span', { className: 'coveo-simplefilter-selecttext' }, l(this.options.title));
+    const icon = $$('span', { className: 'coveo-icon' });
+
+    this.checkboxContainer = $$('div', { className: 'coveo-checkbox-container' });
+    this.checkboxes = _.map(this.options.values, (value) => this.createCheckboxes(value));
+    _.each(this.checkboxes, (result) => {
+      this.checkboxContainer.el.appendChild(result.checkbox.getElement());
+    });
+
+    select.el.appendChild(selectText.el);
+    select.el.appendChild(this.buildCircleElement().el);
+    select.el.appendChild(icon.el);
+    this.element.appendChild(select.el);
+    this.element.appendChild(this.checkboxContainer.el);
+    this.createWrapper().append(this.element);
+
+    $$(this.element).on('click', (e: Event) => this.handleClick(e));
+    this.bind.onRootElement(BreadcrumbEvents.populateBreadcrumb, (args: IPopulateBreadcrumbEventArgs) => this.handlePopulateBreadcrumb(args));
+    // this.bind.onRootElement(BreadcrumbEvents.clearBreadcrumb, () => this.setStateEmpty());
+    this.bind.onRootElement(QueryEvents.buildingQuery, (args: IBuildingQueryEventArgs) => this.handleBuildingQuery(args));
+    this.bind.onRootElement(QueryEvents.querySuccess, (args: IQuerySuccessEventArgs) => this.groupBy(args));
+    this.bind.onRootElement(QueryEvents.doneBuildingQuery, (args: IDoneBuildingQueryEventArgs) => this.handleDoneBuildingQuery(args));
+  }
+
+  private handleBuildingQuery(args: IBuildingQueryEventArgs) {
+    Assert.exists(args);
+    Assert.exists(args.queryBuilder);
+    const selectedValues = this.getSelectedValues();
+    if (selectedValues.length > 0) {
+      args.queryBuilder.advancedExpression.addFieldExpression(this.options.field.toLocaleString(), '==', selectedValues);
+    }
+  }
+
+  public toggle() {
+    this.expanded = $$(this.checkboxContainer).hasClass('coveo-checkbox-container-expanded');
+    if (!this.expanded) {
+      this.open();
+    } else if (this.expanded) {
+      this.close();
+    }
+  }
+
+  private handleClick(e: Event) {
+    if (e.target == this.element || e.target == $$(this.element).find('.coveo-simplefilter-select') ||
+      e.target == $$(this.element).find('.coveo-circle') || e.target == $$(this.element).find('.coveo-icon')
+      || e.target == $$(this.element).find('.coveo-simplefilter-selecttext')) {
+
+      e.stopImmediatePropagation();
+      this.toggle();
+    }
+  }
+
+
+  private open() {
```

Comment:
Those kind of "generic" utilitarian method should be public.

Any "end user" action should be fronted as a public method that is interactable with javascript code.

This will makes the UT much more simpler (because then you will only need to call those public method to test your component).

This also has the bonus that external custom code that wish to interact with the component will be able to, instead of relying on "simulating" click event for example.